kernel/kernel-api/src/main/java/io/delta/kernel/types/StructField.java

```diff
@@ -102,6 +105,47 @@ public String toString() {
         "StructField(name=%s,type=%s,nullable=%s,metadata=%s)", name, dataType, nullable, metadata);
   }
 
+  public FieldMetadata getSerializationMetadata() {
```

Comment:
Why do we need this and how is it different from the `getMetadata`?

Reply:
If I understand correctly, this is capturing the nested field collation types and returning in FieldMetadata. Why is this not already the case when this StructField is created?

Reply:
@stefankandic is this how Spark does? This seems not clear. What is the difference between `getMetadata` vs this method? I understand this has the additional metadata, but for developers I see this causing ambiguity.